framework/include/markers/ReporterPointMarker.h

```diff
@@ -0,0 +1,42 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "Marker.h"
+#include "ReporterInterface.h"
+
+/**
+ * Marks all elements near a given boundary for refinement/coarsening
+ */
+class ReporterPointMarker : public Marker, public ReporterInterface
+{
+public:
+  static InputParameters validParams();
+  ReporterPointMarker(const InputParameters & parameters);
+  virtual void timestepSetup() override;
+
+protected:
+  virtual MarkerValue computeElementMarker() override;
+
+  /// marker value to give elements containing a point
+  const MarkerValue _inside;
+  /// marker for elements not containing points
+  const MarkerValue _empty;
+  /// x coordinate
+  const std::vector<Real> & _x_coord;
+  /// y coordinate
+  const std::vector<Real> & _y_coord;
+  ///z coordinate
+  const std::vector<Real> & _z_coord;
+  /// Pointer to PointLocatorBase object
+  std::unique_ptr<PointLocatorBase> _pl;
+  /// list of sort uniqued elements containing points
```

Comment:
Why not dof IDs?

Reply:
What's wrong with elements? It's all local addresses anyway

Reply:
I like ids better.  I don't know how the elem* would be sorted.